From e0e5b36b532f8f53c0e8668dbe2911d0d9aff4ec Mon Sep 17 00:00:00 2001
From: Surej Joseph <Joseph_Surej@shure.com>
Date: Thu, 12 Mar 2020 09:35:36 -0500
Subject: [PATCH] FOS-QEMU: Load uboot proper from bootloader_a partition

This patch supports load uboot proper as a raw image from
bootloader_a partition. In future we will add support to
read current slot from environment variable and select
appropriate bootloader_<> partition.

Disabling few configs in defconfig that was causing PCI ROM
issues.
---
 configs/foundationos_qemux86_mmc_defconfig | 11 +++++++----
 include/configs/qemu-x86.h                 |  4 ++--
 2 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/configs/foundationos_qemux86_mmc_defconfig b/configs/foundationos_qemux86_mmc_defconfig
index 27e3cbaf08..81be2e2db5 100755
--- a/configs/foundationos_qemux86_mmc_defconfig
+++ b/configs/foundationos_qemux86_mmc_defconfig
@@ -38,9 +38,7 @@ CONFIG_CONSOLE_SCROLL_LINES=5
 CONFIG_SPL=y
 CONFIG_SPL_PCI=y
 CONFIG_SPL_DM=y
-CONFIG_SPL_FAT_SUPPORT=y
 CONFIG_SPL_OF_CONTROL=y
-CONFIG_SPL_LOAD_FIT=y
 CONFIG_SPL_ENV_SUPPORT=y
 CONFIG_SPL_LIBCOMMON_SUPPORT=y
 CONFIG_SPL_LIBGENERIC_SUPPORT=y
@@ -52,8 +50,8 @@ CONFIG_SPL_TIMER=y
 CONFIG_SPL_RTC_SUPPORT=y
 CONFIG_SPL_EFI_PARTITION=y
 CONFIG_SPL_DM_RTC=y
+CONFIG_SPL_LEGACY_IMAGE_SUPPORT=y
 # CONFIG_SPL_RAW_IMAGE_SUPPORT is not set
-# CONFIG_SPL_LEGACY_IMAGE_SUPPORT is not set
 
 CONFIG_EFI_PARTIION=y
 CONFIG_PARTITION_TYPE_GUID=y
@@ -70,6 +68,11 @@ CONFIG_CMD_GPT=y
 # CONFIG_CMD_NFS is not set
 
 CONFIG_MMC=y
-CONFIG_DM_MMC=y
 CONFIG_MMC_PCI=y
 CONFIG_MMC_SDHCI=y
+
+
+CONFIG_DM_MMC=y
+# CONFIG_DM_VIDEO is not set
+
+# CONFIG_VIDEO_VESA is not set
diff --git a/include/configs/qemu-x86.h b/include/configs/qemu-x86.h
index cd1f4c70b9..d88a857a87 100644
--- a/include/configs/qemu-x86.h
+++ b/include/configs/qemu-x86.h
@@ -39,8 +39,8 @@
 
 #define BOOT_DEVICE_SPI			10
 
-#define CONFIG_SYS_MMCSD_FS_BOOT_PARTITION 2
-#define CONFIG_SPL_FS_LOAD_PAYLOAD_NAME "u-boot.img"
+#define CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_USE_PARTITION
+#define CONFIG_SYS_MMCSD_RAW_MODE_U_BOOT_PARTITION 2
 #define BOOT_DEVICE_BOARD		11
 
 
-- 
2.17.1

